src/Plugin/resource/DataProvider/CacheDecoratedDataProvider.php

```diff
@@ -46,6 +46,20 @@ public function __construct(DataProviderInterface $subject, \DrupalCacheInterfac
   /**
    * {@inheritdoc}
    */
+  public static function isNestedField($field_name) {
```

Comment:
So you can inherit from different parents, and decide it at runtime.

If you need to alter (for instance) `entityCheckAccess` for your resource you need to:

  - Override the `Resource::dataProviderClassName` to return a custom data provider class name `\Drupal\…\DataProviderNoAccessCheck`.
  - Implement the new `class DataProviderNoAccessCheck extends DataProviderNode` and override `entityCheckAccess`.

At that point, you cannot add caching to that since your class extends from `DataProviderNode`, and not the hypothetical `class DataProviderCached extends DataProviderBase`.